From d25ff4c614c64f0db22caae60c28ead1ada39324 Mon Sep 17 00:00:00 2001
From: Brandon Hurst <brandon.hurst@analog.com>
Date: Tue, 17 Feb 2026 10:56:36 -0800
Subject: [PATCH 33/39] drivers: usb: udc: max32: Minor function movements

Move xfer_in_done and xfer_out_done to after xfer_in
and xfer_out. This is because they will need to call
these functions in the next commit to handle deferred
traffic.

Signed-off-by: Brandon Hurst <brandon.hurst@analog.com>
---
 drivers/usb/udc/udc_max32.c | 113 ++++++++++++++++++------------------
 1 file changed, 58 insertions(+), 55 deletions(-)

diff --git a/drivers/usb/udc/udc_max32.c b/drivers/usb/udc/udc_max32.c
index b083f53314f..9897cc179a9 100644
--- a/drivers/usb/udc/udc_max32.c
+++ b/drivers/usb/udc/udc_max32.c
@@ -115,62 +115,8 @@ static void udc_event_xfer_out_callback(void *cbdata)
 }
 
 /*
- * Thread-context (non ISR-safe) handlers for xfer completions.
+ * Thread-context handlers for triggering xfer in / out
  */
-static void udc_event_xfer_in_done(const struct device *dev, struct udc_ep_config *ep_cfg)
-{
-	struct udc_max32_data *priv = udc_get_private(dev);
-	MXC_USB_Req_t *ep_request = &priv->ep_request[USB_EP_GET_IDX(ep_cfg->addr)];
-	struct net_buf *buf;
-
-	buf = udc_buf_get(ep_cfg);
-
-	udc_ep_set_busy(ep_cfg, false);
-
-	if (ep_request->error_code) {
-		LOG_ERR("ep 0x%02x error: %x", ep_cfg->addr, ep_request->error_code);
-		udc_submit_ep_event(dev, buf, ep_request->error_code);
-		return;
-	}
-
-	if (udc_ep_buf_has_zlp(buf)) {
-		udc_ep_buf_clear_zlp(buf);
-	}
-
-	if (ep_cfg->addr == USB_CONTROL_EP_IN) {
-		udc_event_xfer_ctrl_status(dev, buf);
-	} else {
-		udc_submit_ep_event(dev, buf, 0);
-	}
-}
-
-static void udc_event_xfer_out_done(const struct device *dev, struct udc_ep_config *ep_cfg)
-{
-	struct udc_max32_data *priv = udc_get_private(dev);
-	MXC_USB_Req_t *ep_request = &priv->ep_request[USB_EP_GET_IDX(ep_cfg->addr)];
-	struct net_buf *buf;
-
-	buf = udc_buf_get(ep_cfg);
-	net_buf_add(buf, ep_request->actlen);
-
-	udc_ep_set_busy(ep_cfg, false);
-
-	if (ep_request->error_code) {
-		LOG_ERR("ep 0x%02x error: %x", ep_cfg->addr, ep_request->error_code);
-		udc_submit_ep_event(dev, buf, ep_request->error_code);
-		return;
-	}
-
-	if (ep_cfg->addr == USB_CONTROL_EP_OUT) {
-		/* Update to next stage of control transfer */
-		udc_ctrl_update_stage(dev, buf);
-
-		udc_ctrl_submit_s_out_status(dev, buf);
-	} else {
-		udc_submit_ep_event(dev, buf, 0);
-	}
-}
-
 static void udc_event_xfer_in(const struct device *dev, struct udc_ep_config *ep_cfg)
 {
 	struct udc_max32_data *priv = udc_get_private(dev);
@@ -259,6 +205,63 @@ static void udc_event_xfer_out(const struct device *dev, struct udc_ep_config *e
 	}
 }
 
+/*
+ * Thread-context (non ISR-safe) handlers for xfer completions.
+ */
+static void udc_event_xfer_in_done(const struct device *dev, struct udc_ep_config *ep_cfg)
+{
+	struct udc_max32_data *priv = udc_get_private(dev);
+	MXC_USB_Req_t *ep_request = &priv->ep_request[USB_EP_GET_IDX(ep_cfg->addr)];
+	struct net_buf *buf;
+
+	buf = udc_buf_get(ep_cfg);
+
+	udc_ep_set_busy(ep_cfg, false);
+
+	if (ep_request->error_code) {
+		LOG_ERR("ep 0x%02x error: %x", ep_cfg->addr, ep_request->error_code);
+		udc_submit_ep_event(dev, buf, ep_request->error_code);
+		return;
+	}
+
+	if (udc_ep_buf_has_zlp(buf)) {
+		udc_ep_buf_clear_zlp(buf);
+	}
+
+	if (ep_cfg->addr == USB_CONTROL_EP_IN) {
+		udc_event_xfer_ctrl_status(dev, buf);
+	} else {
+		udc_submit_ep_event(dev, buf, 0);
+	}
+}
+
+static void udc_event_xfer_out_done(const struct device *dev, struct udc_ep_config *ep_cfg)
+{
+	struct udc_max32_data *priv = udc_get_private(dev);
+	MXC_USB_Req_t *ep_request = &priv->ep_request[USB_EP_GET_IDX(ep_cfg->addr)];
+	struct net_buf *buf;
+
+	buf = udc_buf_get(ep_cfg);
+	net_buf_add(buf, ep_request->actlen);
+
+	udc_ep_set_busy(ep_cfg, false);
+
+	if (ep_request->error_code) {
+		LOG_ERR("ep 0x%02x error: %x", ep_cfg->addr, ep_request->error_code);
+		udc_submit_ep_event(dev, buf, ep_request->error_code);
+		return;
+	}
+
+	if (ep_cfg->addr == USB_CONTROL_EP_OUT) {
+		/* Update to next stage of control transfer */
+		udc_ctrl_update_stage(dev, buf);
+
+		udc_ctrl_submit_s_out_status(dev, buf);
+	} else {
+		udc_submit_ep_event(dev, buf, 0);
+	}
+}
+
 static int udc_ctrl_feed_dout(const struct device *dev, const size_t length)
 {
 	struct udc_max32_data *priv = udc_get_private(dev);
-- 
2.47.3

